frontend/app/routes/settings/account-settings.tsx

```diff
@@ -1,14 +1,173 @@
+    first_name: formData.get("first_name")?.toString() || "",
+    last_name: formData.get("last_name")?.toString() || ""
+  } satisfies RequestInput<"patch", "/api/auth/update-profile/">;
+
+  const { error: errors } = await apiClient.PATCH("/api/auth/update-profile/", {
+    headers: await getCsrfTokenHeader(),
+    body: profileData
+  });
+
+  if (errors) return { success: false, errors };
+
+  queryClient.invalidateQueries(userQueries.authedUser);
+
+  toast.success("Success", {
+    description: "Profile updated successfully",
+    closeButton: true
+  });
+
+  return { success: true };
 }
 
 export default function UserSettingsPage({}: Route.ComponentProps) {
   return (
-    <>
-      <h2 className="text-grey italic text-center">✨ Coming soon ✨</h2>
-    </>
+    <section className="flex flex-col gap-4">
+      <div className="flex items-center gap-4">
+        <h2 className="text-2xl">Profile</h2>
+      </div>
+      <Separator />
+      <p className="text-grey">Update your profile information</p>
+      <div className="grid lg:grid-cols-12 gap-10 relative">
+        <div className="lg:col-span-10 flex flex-col">
+          <section id="update-profile" className="flex gap-1 scroll-mt-20">
+            <div className="w-16 hidden md:flex flex-col items-center">
+              <div className="flex rounded-full size-10 flex-none items-center justify-center p-1 border-2 border-grey/50">
+                <UserIcon size={15} className="flex-none text-grey" />
+              </div>
+              <div className="h-full border border-grey/50"></div>
+            </div>
+            <div className="w-full flex flex-col gap-5 pt-1 pb-8">
+              <h3 className="text-lg text-grey">Profile Information</h3>
+              <UpdateProfileForm />
+            </div>
+          </section>
+
+          <section id="update-password" className="flex gap-1 scroll-mt-20">
+            <div className="w-16 hidden md:flex flex-col items-center">
+              <div className="flex rounded-full size-10 flex-none items-center justify-center p-1 border-2 border-grey/50">
+                <KeyIcon size={15} className="flex-none text-grey" />
+              </div>
+            </div>
+            <div className="w-full pt-1 pb-8 flex flex-col gap-2">
+              <h3 className="text-lg text-grey">Change Password</h3>
+              <div>
+                <Link
+                  to="/settings/account/change-password"
+                  className="hover:underline text-sm py-2 flex items-center gap-0.5 text-link"
+                >
+                  <span>Change your password here</span>
+                  <ChevronRight size={15} />
+                </Link>
+              </div>
+            </div>
+          </section>
+        </div>
+      </div>
+    </section>
+  );
+}
+
+function UpdateProfileForm() {
+  const fetcher = useFetcher<typeof clientAction>();
+  const { data: userData } = useQuery(userQueries.authedUser);
```

Comment:
```suggestion
  const { data: user } = useQuery({
  	...userQueries.authedUser,
  	initialData: loaderData.user
  });
```

please preload the user from the `clientLoader` to not have to check for `undefined` user.

Here is an example of how we preload the data in the `clientLoader`: 

https://github.com/zane-ops/zane-ops/blob/d3783539105a019eebbbb006ee20a1efa945d8c0/frontend/app/routes/projects/preview-template-details.tsx#L86-L94

And we use `initialData` for the query, no need to handle for userData?.data and such:

https://github.com/zane-ops/zane-ops/blob/d3783539105a019eebbbb006ee20a1efa945d8c0/frontend/app/routes/projects/preview-template-details.tsx#L105-L108